From 5046d54855bb9c4f683516fc9a1d1107e89a84ea Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@debian.org>
Date: Sat, 12 Jan 2013 14:00:24 +0100
Subject: [PATCH] Remove explicit field exporting rules from those understood
 by dpkg-dev

These have been supported for a long time now, since 2007.
---
 doc/devel/historic/kernel-modules.txt          | 4 ++--
 doc/devel/internals/udebs.xml                  | 8 ++++----
 doc/devel/modules.txt                          | 2 +-
 doc/talks/d-i_debconf6/d-i_debconf6.tex        | 6 +++---
 doc/talks/d-i_debconf6/slides/d-i_debconf6.tex | 4 ++--
 5 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/doc/devel/historic/kernel-modules.txt b/doc/devel/historic/kernel-modules.txt
index 9280325..9efb21e 100644
--- a/doc/devel/historic/kernel-modules.txt
+++ b/doc/devel/historic/kernel-modules.txt
@@ -9,9 +9,9 @@ kernel modules for the running kernel, so anna needs a way to figure
 out
   1. Whether a udeb is a kernel modules udeb
   2. What kernel versino it belongs to
-To solve this, we introduce the  XB-kernel-version  control field. It
+To solve this, we introduce the  Kernel-Version  control field. It
 will simply say
-  XB-kernel-version: 2.4.19
+  Kernel-Version: 2.4.19
 for the 2.4.19 kernel module udebs, et.c.
 
 The memory footprint is small, on i386 I think installing all available
diff --git a/doc/devel/internals/udebs.xml b/doc/devel/internals/udebs.xml
index 664d600..bc64841 100644
--- a/doc/devel/internals/udebs.xml
+++ b/doc/devel/internals/udebs.xml
@@ -129,17 +129,17 @@ Build-Depends: debhelper (&gt;= 7.3.10), libdebian-installer4-dev (&gt;= 0.41),
 
 Package: kbd-chooser
 Architecture: i386 amd64 powerpc alpha hppa sparc [...]
-XC-Package-Type: udeb
+Package-Type: udeb
 Depends: ${shlibs:Depends}, ${misc:Depends}, console-keymaps
 Description: Detect a keyboard and select layout
-XB-Installer-Menu-Item: 1200
+Installer-Menu-Item: 1200
 </screen></informalexample>
 
 <para>
 
-The line <literal>XC-Package-Type</literal> tells
+The line <literal>Package-Type</literal> tells
 <classname>debhelper</classname> to treat the package as a udeb. The
-<literal>XB-Installer-Menu-Item</literal> is added in the control file for
+<literal>Installer-Menu-Item</literal> is added in the control file for
 the udeb and will eventually end up in the <classname>dpkg</classname>
 status file to help <classname>main-menu</classname> figure out that this
 udeb should be included in the menu and in what order<footnote>
diff --git a/doc/devel/modules.txt b/doc/devel/modules.txt
index ffd4618..fc3b9f1 100644
--- a/doc/devel/modules.txt
+++ b/doc/devel/modules.txt
@@ -118,5 +118,5 @@ A 2 minute primer on building udebs.
 ====================================
 
 1. Add a Build-Dependency against debhelper (>= 4.2).
-2. Add "XC-Package-Type: udeb" to the package's stanza in the control file.
+2. Add "Package-Type: udeb" to the package's stanza in the control file.
 3. Build the package!
diff --git a/doc/talks/d-i_debconf6/d-i_debconf6.tex b/doc/talks/d-i_debconf6/d-i_debconf6.tex
index f53e88a..0d2c45f 100644
--- a/doc/talks/d-i_debconf6/d-i_debconf6.tex
+++ b/doc/talks/d-i_debconf6/d-i_debconf6.tex
@@ -365,13 +365,13 @@ Build-Depends: debhelper (>= 5.0.22), libdebian-installer4-dev (>= 0.41), po-deb
 
 Package: kbd-chooser
 Architecture: i386 amd64 powerpc alpha hppa sparc [...]
-XC-Package-Type: udeb
+Package-Type: udeb
 Depends: ${shlibs:Depends}, ${misc:Depends}, console-keymaps
 Description: Detect a keyboard and select layout
-XB-Installer-Menu-Item: 12
+Installer-Menu-Item: 12
 \end{verbatim}
 
-The line \texttt{XC-Package-Type} tells debhelper to treat the package as a udeb. The \texttt{XB-Installer-Menu-Item} is added in the control file for the udeb and will eventually end up in the dpkg status file to help main-menu figure out that this udeb should be included in the menu and in what order\footnote{The file \texttt{installer/doc/devel/menu-item-numbers.txt} in the d-i SVN repository documents menu numbers currently in use.}. Packaging a udeb becomes a bit harder if it is derived from a regular package but needs to be compiled with different compiler options (e.g. some features disabled or a different optimization). 
+The line \texttt{Package-Type} tells debhelper to treat the package as a udeb. The \texttt{Installer-Menu-Item} is added in the control file for the udeb and will eventually end up in the dpkg status file to help main-menu figure out that this udeb should be included in the menu and in what order\footnote{The file \texttt{installer/doc/devel/menu-item-numbers.txt} in the d-i SVN repository documents menu numbers currently in use.}. Packaging a udeb becomes a bit harder if it is derived from a regular package but needs to be compiled with different compiler options (e.g. some features disabled or a different optimization). 
 
 The main thing to always keep in mind when creating a udeb is size. It is very important to keep size as minimal as possible. This includes using tabs instead of spaces when indenting in scripts and not being too verbose in comments. 
 
diff --git a/doc/talks/d-i_debconf6/slides/d-i_debconf6.tex b/doc/talks/d-i_debconf6/slides/d-i_debconf6.tex
index 56b1bef..842a226 100644
--- a/doc/talks/d-i_debconf6/slides/d-i_debconf6.tex
+++ b/doc/talks/d-i_debconf6/slides/d-i_debconf6.tex
@@ -424,10 +424,10 @@ Build-Depends: debhelper (>= 5.0.22), libdebian-installer4-dev (>= 0.41),
 
 Package: kbd-chooser
 Architecture: i386 amd64 powerpc alpha hppa sparc [...]
-\color{red}XC-Package-Type: udeb\normalcolor
+\color{red}Package-Type: udeb\normalcolor
 Depends: $\{shlibs:Depends\}, $\{misc:Depends\}, console-keymaps
 Description: Detect a keyboard and select layout
-\color{red}XB-Installer-Menu-Item: 12\normalcolor
+\color{red}Installer-Menu-Item: 12\normalcolor
 \end{semiverbatim}
 \end{tiny}
 \end{frame}
-- 
1.8.1

